File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = 0
        current_sum = 0
        prefixs = {0:1}

        for i in nums:
            current_sum += i
            diff = current_sum - k
            count += prefixs.get(diff, 0)
            prefixs[current_sum] = 1 + prefixs.get(current_sum, 0)
        return count

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> mp;
        int prefixsum=0,ans=0;
        mp[0] = 1;
        for(auto it:nums){
            prefixsum += it;
            int difference = prefixsum - k;
            if(mp.find(difference) != mp.end()){
                ans += mp[difference];
            }
            mp[prefixsum]++;
        }
        return ans;
    }
};